#! /bin/sh /usr/share/dpatch/dpatch-run
## 14_fix_amd64_no_mmx_build.dpatch by Kevin Goodsell <kevin-opensource@omegacrash.net>
##
## DP: Fix building on AMD64. Original patch by Andreas Schultz in LP: #62209.

@DPATCH@
diff -urNad mplayer-trunk~/libvo/osd_template.c mplayer-trunk/libvo/osd_template.c
--- mplayer-trunk~/libvo/osd_template.c	2009-07-14 10:31:01.000000000 -0700
+++ mplayer-trunk/libvo/osd_template.c	2009-07-14 11:05:23.000000000 -0700
@@ -452,7 +452,7 @@
 		"movb %%dh, 2(%0)\n\t"
 
 		:
-		:"r" (&dstbase[4*x]),
+		:"R" (&dstbase[4*x]),
 		 "r" ((unsigned)srca[x]),
 		 "r" (((unsigned)src[x])<<8)
 		:"%eax", "%ecx", "%edx"
